api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -158,13 +159,17 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 					}
 					creationCount++
 				default:
-					catalog.Spec.Name = name
-					catalog.ObjectMeta.Name = name
-					catalog.Spec.SiteId = i.Context.SiteInfo.SiteId
+					// Check wrapped catalog structure and extract wrapped catalog name
 					objectData, _ := json.Marshal(catalog.Spec.Properties)
-					err = utils.UpsertCatalog(ctx, i.Config.BaseUrl, name, i.Config.User, i.Config.Password, objectData)
+					var catalogState model.CatalogState
```

Comment:
I think this whole function needs some work, not just for the catalog branch, but for other object types as well: 1) check the embedded object type; 2) since we anticipate properties can contain metadata and spec, if a metadata is supplied, we need to use information in the metadata - like name and namespaces; 3) add unit test cases for all object types;

Reply:
Fixed.